src/inferred_mode/inferred_mode.ts

```diff
@@ -69,7 +67,6 @@ export default class InferredQueryHandler {
     this.path = path;
     this.predicatePath = predicatePath;
     this.includeReasoner = includeReasoner;
-    this.CREATIVE_LIMIT = process.env.CREATIVE_LIMIT ? parseInt(process.env.CREATIVE_LIMIT) : 500;
```

Comment:
Creative limit should be kept, but used only for truncation, rather than as a stopping threshold